tmt/steps/provision/connect.py

```diff
@@ -132,19 +132,30 @@ def reboot(
             )
 
         if command is not None:
+            reboot_command = command
+
+            if str(command) == str(tmt.steps.DEFAULT_REBOOT_COMMAND) and self.become:
+                reboot_command = ShellScript(f'sudo {reboot_command}')
+
             return super().reboot(
                 hard=False,
-                command=command,
+                command=reboot_command,
                 waiting=waiting,
             )
 
         if self.soft_reboot is not None:
-            self.debug(f"Soft reboot using the soft reboot command '{self.soft_reboot}'.")
+            reboot_command = self.soft_reboot
```

Comment:
This does not look correct to me: user provided a reboot command, and tmt should blindly use it. If it fails without `sudo`, it fails and reboot would not happen, but that's user fault. tmt should take the provided command and execute it. It may be a custom script containing complex command, a script already whitelisted with sudoers, for example.

Reply:
indeed, in the case of my reproducer the `self.soft_reboot` is not set. Reverted